src/metabase/models/data_permissions.clj

```diff
@@ -434,9 +438,27 @@
    value       :- :keyword]
   (set-table-permissions! group-or-id perm-type {table-or-id value}))
 
+(defn- schema-permission-value
+  "Infers the permission value for a new table based on existing permissions in the schema.
+  Returns the uniform permission value if one exists, otherwise nil."
+  [db-id group-id schema-name perm-type]
+  (let [possible-values    (:values (get Permissions perm-type))
+        schema-perms-check (mapv (fn [value]
+                                   (t2/exists? :model/DataPermissions
+                                               :perm_type   (u/qualified-name perm-type)
+                                               :db_id       db-id
+                                               :group_id    group-id
+                                               :schema_name schema-name
+                                               :perm_value  value))
+                                 possible-values)
```

Comment:
We might want to use a sql statement with :in so we send fewer queries.

Reply:
Well, we specifically want to know whether any row exists in the schema for each individual possible value. That's why there's a separate `t2/exists?` call for each value. If rows only exist for one possible value, we know that tables in the schema are set consistently; otherwise, we know that they're set granularly.

If we did something like `[:in perm_value possible-values]` we'd know whether _any_ rows exist for tables in the schema, but wouldn't know whether they're all the same or not.

Does that make sense? Maybe there's a more complicated query that could do this in a single call but I'm not totally sure how it would work.

Reply:
Yeah that might be a premature optimization, given that there are only 2 or 3 values anyway. We could revisit if it's actually an issue.